drivers/spi/spi_rpi_pico_pio.c

```diff
@@ -651,6 +572,8 @@ static void spi_pico_pio_txrx_3_wire(const struct device *dev)
 		}
 	}
 
+	k_sleep(K_TICKS(data->tx_period_ticks));
```

Comment:
Why is this needed?

Reply:
After the FIFO is drained and the state machine completes sending, the peripherals I was testing with (a BME280 and a TI TMAG5170 magnetometer) weren't getting the last bit.  The logic analyzer trace showed the GPIO line was changing state before that last bit was latched.  One more SPI clock period was needed before resetting the GPIO line for read mode.

Can you see a better place to insert that delay?

Reply:
I'm just always suspicious of delays in driver code. Ideally the exact timings of the GPIO line should be handled by the PIO, and the CPU should just wait for the PIO to finish (I'm also doubtful that the tick resolution is high enough to actually produce a delay of the correct length).
That said, if you insist that this delay is necessary, please just add a short comment in the code explaining why the delay is needed - in a hypothetical future where we'll be debugging the timing of this code, finding this undocumented delay would be an unpleasant surprise :)